presto-pinot/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -0,0 +1,201 @@
+    {
+        final String url = String.format(TABLE_SCHEMA_API_TEMPLATE, getControllerUrl(), table);
+        String responseBody = sendHttpGet(url);
+        return Schema.fromString(responseBody);
+    }
+
+    public String getBrokerHost(String table)
+            throws Exception
+    {
+        if (dynamicBrokerSelector == null) {
+            dynamicBrokerSelector = new DynamicBrokerSelector(zookeeperServers);
+        }
+        return this.dynamicBrokerSelector.selectBroker(table);
+    }
+
+    public Map<String, Map<String, List<String>>> getRoutingTableForTable(String tableName)
+            throws Exception
+    {
+        final Map<String, Map<String, List<String>>> routingTableMap = new HashMap<>();
+        final String url = String.format(ROUTING_TABLE_API_TEMPLATE, getBrokerHost(tableName), tableName);
+        String responseBody = sendHttpGet(url);
+        log.debug("Trying to get routingTable for %s. url: %s", tableName, url);
+        JSONObject resp = JSONObject.parseObject(responseBody);
+        JSONArray routingTableSnapshots = resp.getJSONArray("routingTableSnapshot");
+        for (int i = 0; i < routingTableSnapshots.size(); i++) {
+            JSONObject snapshot = routingTableSnapshots.getJSONObject(i);
+            String tableNameWithType = snapshot.getString("tableName");
+            // Response could contain info for tableName that matches the original table by prefix.
+            // e.g. when table name is "table1", response could contain routingTable for "table1_staging".
+            // The bug is fixed in the latest Pinot but not in the release version 0.1.0
+            if (!tableName.equals(extractRawTableName(tableNameWithType))) {
+                log.debug("Ignoring routingTable for %s", tableNameWithType);
+                continue;
+            }
+            JSONArray routingTableEntriesArray = snapshot.getJSONArray("routingTableEntries");
+            if (routingTableEntriesArray.size() == 0) {
+                throw new PrestoException(
+                        PINOT_FAILURE_GETTING_TABLE,
+                        "RoutingTable is empty for " + tableName);
+            }
+            String routingTableEntries = routingTableEntriesArray.getJSONObject(new Random().nextInt(routingTableEntriesArray.size())).toJSONString();
+            Map<String, List<String>> routingTable = objectMapper.readValue(routingTableEntries, Map.class);
+            routingTableMap.put(tableNameWithType, routingTable);
+        }
+        return routingTableMap;
+    }
+
+    public Map<String, String> getTimeBoundaryForTable(String table)
+            throws Exception
+    {
+        final String url = String.format(TIME_BOUNDARY_API_TEMPLATE, getBrokerHost(table), table);
+        String responseBody = sendHttpGet(url);
+        JSONObject resp = JSONObject.parseObject(responseBody);
+        Map<String, String> timeBoundary = new HashMap<>();
+        if (resp.containsKey("timeColumnName")) {
```

Comment:
These strings should be public constants. I see them appear in other classes as well.